### DISCUSSION TITLE
Any non dev looking displays using tasmota?

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#20775] [Q&A] [answered]
I've looked through all of the hardware supported but have not found a display that could be placed in a living room for example, that doesn't look like a development board but has a nice complete case.

Vertical or horizontal, it doesn't matter much, I just want to display a few lines of text or very basic images. 

Does anyone know of a nice looking display that is already tasmota flashed or will always be flashable?

**Comment by barbudor** [comment]
Most common ones
https://templates.blakadder.com/lanbon_L8-HS.html
https://templates.blakadder.com/sonoff_NSPanel.html

Not using them myself
But today it's not so difficult to give a pro look to a DIY stuff

**Comment by dac2020** [comment]
Hi,

Thank you for the leads. 
By a pro look, do you mean using 3d printing or something, which I don't have.
I'm trying to find something that can sit on a shelf for example, that doesn't have wires coming out all over.

**Reply by sfromis** [reply, marked as answer]
There's also a "shelf-ready" Tasmota-compatible device with a display:
https://templates.blakadder.com/espressif_ESP32-S3-BOX.html
Intended for development, but can work stand-alone, just with USB power.

Smaller ones coming in a box also exists, like 
https://templates.blakadder.com/m5stack_coreS3.html

When it comes to 3D print, there are makerlabs many places, where people can come to use a 3D printer and other DIY toys. You could probably also get someone to help you, if you already have found a project you want to print.

**Comment by dac2020** [comment]
Hi,

Thanks for the leads also.

The links to Amazon and AliExpress both show currently not available.

But I found this;
https://www.amazon.com/Espressif-ESP32-S3-BOX-3B-Development-Board/dp/B0CSJC5BX6/
It seems to be the same thing but the listing is not very clear. It's hard to know if it is the display and the stand or only the stand.
Maybe I'll order one however it seems to be a hit and miss availability.

**Reply by sfromis** [reply]
When viewing the template pages, do not see the links as "the place" to buy the devices, availability can indeed be pretty mixed. Instead search for the device name.

Anyway, the page you found does have an image to indicate that both display, dock, cable and a led module is included. The blue variant has additional stuff.
> The ESP32-S3-BOX-3 offers two different versions: ESP32-S3-BOX-3 and ESP32-S3-BOX-3B. The ESP32-S3-BOX-3 represents the standard edition with four blue accessories, the ESP32-S3-BOX-3B provides fewer accessories to meet starters who don’t need much unnecessary hardware.

**Comment by dac2020** [comment]
So I received the esp32 box and before I tasmota this thing, I wanted to ask this.
Once it's converted, will I be able to display images or only text? I've never played with a display before and just want to make sure I don't mess this up. 

I have to find a doc or article, some hints on doing small things. I found the 'display' page for tasmota but it's overwhelming for first time trying with this.

**Reply by sfromis** [reply]
Yes, you should be able to also display images, and use downloadable fonts. And make much nicer display design than simple text.

What I'd recommend would be to config it for [Universal Display](https://tasmota.github.io/docs/Displays/#universal-display-driver) (already done below), and use [HASPmota](https://tasmota.github.io/docs/HASPmota/) or [LVGL](https://tasmota.github.io/docs/LVGL/), instead of the old-style display commands.

First you need a relevant binary, probably something like `tasmota32s3cdc-qio_opi.factory.bin` from
https://github.com/tasmota/install/tree/firmware/firmware/unofficial which you can select in the [Tasmota Web Installer](https://tasmota.github.io/install/) as "Tasmota32-S3 (QIO Flash/OPI PSRAM). That image includes support for LVGL and HASPmota, and I'd expect it to enable PSRAM for your device. Next, run Auto-configuration selecting "BOX". This installs a template including enabling Universal Display, and the needed `display.ini` file.

**Comment by dac2020** [comment]
Nice, then I'll give it a try. 
The template doc mentions this;

>Uses an [ESP32-S3-WROOM-1-N16R8](https://www.espressif.com/sites/default/files/documentation/esp32-s3->wroom-1_wroom-1u_datasheet_en.pdf) module with 16 MB flash and 8 MB PSRAM.

If I understand, it's not saying I need something else, just that this device is using the above right? Just want to make sure I understand before going ahead here.

Moving on, I get this when trying to download the file. Maybe someone at tasmota needs to know about this?
![image](https://github.com/arendst/Tasmota/assets/9727841/e5045596-830c-49ea-b98a-538d5cd9baa7)

**Reply by dac2020** [reply]
I tried a few times and it finally worked so off I go converting this.

**Reply by sfromis** [reply]
I see no issue in the screenshot you posted. That's just how Github works. Anyway, I was deliberately *not* recommending you to download, but instead use the Tasmota Web Installer; still you *can* download via the links to the right in the screenshot.

And yes, I suspect that module, that's why I made the specific suggestion for qio/opi.

**Comment by dac2020** [comment]
I always like to have a copy which is why I wanted to download it but yes, I see it in the tasmota installer.
I've converted the box to tasmota now. Thanks for the help.

**Comment by dac2020** [comment]
Done, without a problem of course which is why I love Tasmota!
I used the auto configure as you suggested which created a template but the template page shows a different one.

{"NAME":"ESP-S3-BOX","GPIO":[0,160,0,0,6624,6592,864,896,640,1,1,1,1,1,1,0,0,0,608,1,1,1,0,0,0,0,0,1,1,1,1,1,1,1,992,0,0,1024],"FLAG":0,"BASE":1}

Which should I be using?

As the template page suggests, I ran
DisplayHeight 240
DisplayWidth 320

Now, I can't wait to put an image on it or some text. 
Does anyone know of a url that might give me some examples of how I can do that?

**Comment by s-hadinger** [comment]
I strongly suggest to use HASPmota. You get a rich set of features 
https://tasmota.github.io/docs/HASPmota/
